### PR TITLE
Refs #6853 String template issues fiexd.

### PR DESCRIPTION
**Pull request by LuisGP**
This PR fixes some StringTemplate methods used by Websocket.

### CODE REVIEW

File: packages/core/src/FieldToString.cpp [hunk 1]
```diff
@@ -51,38 +51,38 @@ class FieldConversions
...
-    add_primitive_conversion<int16_t>();
-    add_primitive_conversion<uint16_t>();
-    add_primitive_conversion<int32_t>();
-    add_primitive_conversion<uint32_t>();
-    add_primitive_conversion<int64_t>();
-    add_primitive_conversion<uint64_t>();
-    add_primitive_conversion<float>();
-    add_primitive_conversion<double>();
-    add_primitive_conversion<long double>();
+    add_primitive_conversion<bool>("bool");
```
Comment (lemunozm) [inline packages/core/src/FieldToString.cpp]:
```suggestion
    add_primitive_conversion<bool>(xtypes::PrimitiveType<bool>::Name);
```
Check if works.

File: packages/core/src/StringTemplate.cpp [hunk 2]
```diff
@@ -81,7 +81,8 @@ class StringTemplate::Implementation
           throw UnavailableMessageField(field_name, converter.details);
         }
 
-        const xtypes::DynamicData& data = message.cref()[field_name];
+        //xtypes::ReadableDynamicDataRef data = message.cref()[field_name];
```
[the remark this answers is not in the document]
Comment (LuisGP) [reply packages/core/src/StringTemplate.cpp]: Removed

File: packages/core/src/StringTemplate.cpp [hunk 3]
```diff
@@ -66,10 +66,10 @@ class StringTemplate::Implementation
 
   std::string compute_string(const xtypes::DynamicData& message) const
   {
-    std::string result;
+    std::string result = components[0];
```
[the remark this answers is not in the document]
Comment (LuisGP) [reply packages/core/src/StringTemplate.cpp]: IDK. Added a check just in case.

File: packages/core/src/FieldToString.cpp (hunk 1 above)
Comment (LuisGP) [reply packages/core/src/FieldToString.cpp]: Following your idea, I change it reverting most of the changes. Please, take a look.

**Pull request merged**